### ISSUE TITLE
Problem with training file in args.py and the demographic label representation

### ISSUE THREAD

**Issue by eesss34690** [gongsixue/GAC#1]
Hi, I am currently studying the paper and try to run this code, but there are some questions I encountered.
1. In the paper, the training set is **BUPT-Balancedface** and the testing set is **RFW**, but in args.py, it appears that the dataset for training is **RFW** and the testing set is **rfw_test_Black**. Is this for other purpose or simple demonstration?
```
dataset_train = 'CSVListLoader'
input_filename_train = '/research/prip-gongsixu/datasets/RFW/attr_rfw_balance_aligned_112.txt'
dataset_options_train = {'ifile':input_filename_train, 'ind_attr':[2], 'root':dataset_root_train,
                'transform':preprocess_train, 'loader':loader_input}

// some other code here

dataset_test = 'CSVListLoader'
# dataset_test = 'FileListLoader'
input_filename_test = '/research/prip-gongsixu/datasets/RFW/attr_rfw_test_Black_aligned_112.txt'
# input_filename_test = '/research/prip-gongsixu/datasets/LFW/list_lfw_aligned_retina_112.txt'
dataset_options_test = {'ifile':input_filename_test, 'ind_attr':[2], 'root':dataset_root_test,
                 'transform':preprocess_test, 'loader':loader_input}
```
2. In dataset RFW, there is actually no file of ```attr_rfw_balance_aligned_112.txt'``` and ```attr_rfw_test_Black_aligned_112.txt```. Are they the manual label you made? If they are put into CSVListLoader, there should be at least 3 columns for each image. What is for the demographic label in column [2] defined by you? I saw the dictionary for demographic group but has no idea what is the corresponding meaning.
```
self.dict_attr = {'0':0, '1':1, '2': 2, '3':3, '00':0, '01':1, '02':2, '03':3,\
            '10':4, '11':5, '12':6, '13':7}

// some other code here

category = ''
for j in self.ind_attr:
    category += self.data[index][j]
demog_label = int(self.dict_attr[category])
```
Thank you in advance for reading my problem. Hope to hear the news.

**Comment by gongsixue** [comment]
Hi, sorry for the confusion.

To clarify:

1. The file "attr_rfw_balance_aligned_112.txt" is the BUPT-Balanceface, I renamed it and tagged it as rfw for all datasets related to rfw, see my number of classes in my arg.py: nclasses = 28000 # balance. This the total number of subjects in BUPT-Balanceface. While training, I only use black subset in RFW as the validation set to save time for validation.
2. Three columns: Image path, subject ID, and race label;
     The dictionary is the mapping for race labels and race&gender labels.
     For race only --
            0 -- Indian
            1 -- Caucasian
            2 -- Asian
            3 -- African
     For race&gender --
            00 -- Indian Male
            01 -- Caucasian Male
            02 -- Asian Male
            03 -- African Male
            10 -- Indian Female
            11 -- Caucasian Female
            12 -- Asian Female
            13 -- African Female

**Comment by eesss34690** [comment]
Ok. I got the meaning. Thank you for your answer.

**Comment by shweta4patel** [comment]
How have you found out the gender from the dataset? Did you use any classifier, or is it given in the dataset?
